### DISCUSSION TITLE
Как отцентрировать карту по скоплению близлежащих маркеров?

### DISCUSSION THREAD

**Discussion by svk-man** [yandex-maps-unofficial/vue-yandex-maps#257] [Q&A (2.x)] [answered]
Добрый день!

Прошу помочь разобраться.

Ранее был следующий код: он позволял отцентрировать карту по скоплению маркеров в одном городе при инициализации карты и также при изменении текущего города. Версия была установлена следующая: `"vue-yandex-maps": "^0.12.7"

```vue
const initHandler = (newMap: typeof yandexMap) => {
  map = newMap
  recenter()
}

const recenter = () => {
  setTimeout(() => {
    map.setBounds(map.geoObjects.getBounds()).then(() => {
      if (map.getZoom() > 16) { map.setZoom(16) }
    })
  }, 300)
}

watch(currentCity, () => {
  recenter()
})
```

**_Каким образом можно это реализовать в "vue-yandex-maps": "^2.0.1"?_**

Прочитав документацию, я нашёл [промис](https://yandex-maps-unofficial.github.io/vue-yandex-maps/api/yandex.html#ready), который поможет мне отреагировать на инициализацию карты. Каким образом я могу сделать всё остальное?

**Comment by daniluk4000** [comment, marked as answer]
Хаюшки! Кроме geoObjects всё просто, и промис вам не нужен.

```vue
<template>
    <yandex-map v-model="map">...</yandex-map>
</template>

<script setup lang="ts">
import { shallowRef } from 'vue';
import type { YMap } from '@yandex/ymaps3-types';

const map = shallowRef<YMap | null>(null);

const recenter = () => {
  if (!map.value) return;

  //setTimeout это был костыль наверное?
  map.value.setLocation({ bounds: getAllMarkersBounds() }).then(() => {
    if (map.value!.zoom > 16) { map.value!.setLocation({ zoom: 16 }) }
  })
}

watch([map, currentCity], () => {
    recenter();
});
</script>
```

По поводу geoObjects и getAllMarkersBounds. Я не знаю простых способов получить эту информацию в Картах 3.0. Я предлагаю вам посчитать это самостоятельно, используя доступную вам информацию. Пример таких рассчетов есть у меня в кластере: https://github.com/yandex-maps-unofficial/vue-yandex-maps/blob/master/packages/vue-yandex-maps/src/components/modules/clusterer/YandexMapClustererClusters.vue#L82

**Reply by svk-man** [reply]
Добрый день! Спасибо большое за Ваш ответ!

> //setTimeout это был костыль наверное?
Да, верно

Посчитал самостоятельно, с Вашим примером это получилось очень быстро!

**Reply by daniluk4000** [reply]
@svk-man ой, вспомнил. Я добавил 2 новые апишки в библиотеку, которые повторяют функционал из этого компонента, на основе этой дискуссии. Можете пользоваться)

**Reply by svk-man** [reply]
@daniluk4000 супер! спасибо! попробую
